proto/spacemesh/v1/smesher_types.proto

```diff
@@ -128,3 +128,16 @@ message PostSetupStatus {
     PostSetupOpts opts = 3; // setup options previously set by the user
     string error_message = 4; // The error message, if the state is STATE_ERROR
 }
+
+message SmeshingStatusRequest {
+    repeated uint32 epochs = 1; // Epochs to query status for. If empty, returns status for current, next and subsequent.
+}
+
+message SmeshingStatusResponse {
+    string status = 1;
```

Comment:
status should also be a numeric code, not a string. `reason` can be a string (or a code).